flows/runs/run.go

```diff
@@ -163,10 +163,29 @@ func (r *flowRun) SetWebhook(rr *utils.RequestResponse) {
 func (r *flowRun) Extra() utils.JSONFragment         { return r.extra }
 func (r *flowRun) SetExtra(extra utils.JSONFragment) { r.extra = extra }
 
+func (r *flowRun) ExpiresOn() *time.Time { return r.expiresOn }
+func (r *flowRun) ResetExpiration(from *time.Time) {
+	if r.Flow().ExpireAfterMinutes() != nil {
+		if from == nil {
+			now := time.Now().UTC()
+			from = &now
+		}
+
+		expiresAfterMinutes := time.Duration(*r.Flow().ExpireAfterMinutes())
```

Comment:
Ya, maybe we should treat anything less than or equal to 0 as none? (or validate >= 0)